### PR TITLE
324 phs 2 generate pdf of sales order and send it through emailwhats app

### PR DESCRIPTION
**Pull request by Chetan0178**
1.Generating sales order PDF
2..Generating sales Invoice PDF
3.send that PDF via Email or What's app

### CODE REVIEW

File: apps/sales/urls.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
     path('sale_return_order/<str:pk>/', SaleReturnOrdersViewSet.as_view(), name='sales-return-orders-detail-update-delete'),
     path('quick_pack/', QuickPackCreateViewSet.as_view(), name='quickpack-list-create'),
     path('quick_pack/<str:pk>/', QuickPackCreateViewSet.as_view(), name='quickpack-detail-update-delete'),
+    path('document_generator/<str:pk>/<str:document_type>/', DocumentGeneratorView.as_view(), name='generate-sale-order-pdf'),
```
Comment (QodeFarm) [inline apps/sales/urls.py:35]: This should be there in masters right?

File: apps/sales/utils/sales_order_rcpt_word_docx.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    replacements = Cust_data
+    for table in doc.tables:
+        for row in table.rows:
+            for cell in row.cells:
+                for paragraph in cell.paragraphs:
+                    for key, value in replacements.items():
+                        if key in paragraph.text:
+                            paragraph.text = paragraph.text.replace(key, value)
+
+    print()
```
Comment (QodeFarm) [inline apps/sales/utils/sales_order_rcpt_word_docx.py:169]: what is this print?

File: apps/sales/views.py [hunk 3]
```diff
@@ -1456,7 +1455,110 @@ def put(self, request, pk, *args, **kwargs):
         ]
 
         return build_response(1, "Records updated successfully", custom_data, status.HTTP_200_OK)
+    
+#================================================================================================================================
+class DocumentGeneratorView(APIView):
```
Comment (QodeFarm) [inline apps/sales/views.py:1460]: Why is this class in sales view?
Cant we move this to masters? if you want you can make a call to method by passing the required params

File: apps/sales/urls.py (hunk 1 above)
Comment (Chetan0178) [reply apps/sales/urls.py:35]: moved into masters.

File: apps/sales/views.py (hunk 3 above)
Comment (Chetan0178) [reply apps/sales/views.py:1460]: no I'll moved it to masters for better understandings.

Comment (Chetan0178) [reply apps/sales/views.py]: moved into masters

File: apps/sales/utils/sales_order_rcpt_word_docx.py (hunk 2 above)
Comment (Chetan0178) [reply apps/sales/utils/sales_order_rcpt_word_docx.py]: removed!

**Pull request merged**